k8s/supportability/src/collect/logs/loki.rs

```diff
@@ -265,7 +265,11 @@ impl LokiClient {
             };
             is_written = true;
             for msg in result.iter() {
-                write!(log_file, "{msg}")?;
+                if msg.ends_with('\n') {
+                    write!(log_file, "{msg}")?;
+                } else {
+                    writeln!(log_file, "{msg}")?;
+                }
```

Comment:
More readable?
```suggestion
            is_written = true;
            result
                .iter()
                .for_each(|msg| writeln!(log_file, "{}", msg.trim_end())?);
```

Reply:
The closures on for_each can't return error like this, I've kept the for but added the trim_end.